Tests/SentryTests/SentryCrash/SentryThreadInspectorTests.swift

```diff
@@ -54,32 +55,43 @@ class SentryThreadInspectorTests: XCTestCase {
         expect.expectedFulfillmentCount = 10
         
         let sut = self.fixture.getSut(testWithRealMachineContextWrapper: true)
+
         for _ in 0..<10 {
-            
+
             queue.async {
-                let actual = sut.getCurrentThreadsWithStackTrace()
-                
-                // Sometimes during tests its possible to have one thread without frames
-                // We just need to make sure we retrieve frame information for at least one other thread than the main thread
-                var threadsWithFrames = 0
-                
-                for thr in actual {
-                    if (thr.stacktrace?.frames.count ?? 0) >= 1 {
-                        threadsWithFrames += 1
+                let threads = sut.getCurrentThreadsWithStackTrace()
+
+                var threadsWithStackTraceFrames = 0
+
+                for thread in threads {
+
+                    guard let frames = thread.stacktrace?.frames else {
+                        continue
+                    }
+
+                    if frames.count == 0 {
+                        continue
                     }
-                    
-                    for frame in thr.stacktrace?.frames ?? [] {
+
+                    for frame in frames {
                         XCTAssertNotNil(frame.instructionAddress)
                         XCTAssertNotNil(frame.imageAddress)
+
                     }
+
+                    threadsWithStackTraceFrames += 1
                 }
-                
-                XCTAssertTrue(threadsWithFrames > 1, "Not enough threads with frames")
-                
+
+                let percantageWithStacktraceFrames = Double(threadsWithStackTraceFrames) / Double(threads.count)
+
+                // During testing we usually have around 90% to 100%
+                // We choose a bit lower threshold to avoid flaky tests in CI
```

Comment:
`m`: It might be worth the effort to add the reasoning why there are threads with absent stack traces, as you described in the PR documentation, or at least a link to this PR so we can find it again.

Reply:
Yes totally makes sense. Thanks for the hint.